charts/internal/shoot-control-plane/templates/rbac-firewall-policy-controller.yaml

```diff
@@ -0,0 +1,93 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:firewall-policy-controller
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - services
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
```

Comment:
From an architectural point of view I think that the firewall controller is deploying arbitrary things into the cluster feels wrong.
We have the Gardener to orchestrate all the deployments of services that run inside the clusters. I want to suggest a pattern similar to how the [K8s Auditing](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#dynamic-backend) works, such that we can dynamically define logging backends for the firewall controller. This would also allow to run the droptailer somewhere else in the future.

Reply:
I would also prefer that the firewall-policy-controller has only minimal and red-only permissions to api-resources.
We could move these things:
- secret creation to the infrastructure reconcilation where we already create the kubeconfig secret
- deployment to charts/internal/shoot-droptailer